src/libs/arduinoFFT/arduinoFFT.cpp

```diff
@@ -0,0 +1,510 @@
+// 	float mean = 0;
+// 	for (uint16_t i = 0; i < samples; i++)
+// 	{
+// 		mean += vData[i];
+// 	}
+// 	mean /= samples;
+// 	// Subtract the mean from vData
+// 	for (uint16_t i = 0; i < samples; i++)
+// 	{
+// 		vData[i] -= mean;
+// 	}
+// }
+
+// void arduinoFFT::Windowing(uint8_t windowType, uint8_t dir)
+// {// Weighing factors are computed once before multiple use of FFT
+// // The weighing function is symmetric; half the weighs are recorded
+// 	float samplesMinusOne = (float(this->_samples) - 1.0f);
+// 	for (uint16_t i = 0; i < (this->_samples >> 1); i++) {
+// 		float indexMinusOne = float(i);
+// 		float ratio = (indexMinusOne / samplesMinusOne);
+// 		float weighingFactor = 1.0f;
+// 		// Compute and record weighting factor
+// 		switch (windowType) {
+// 		case FFT_WIN_TYP_RECTANGLE: // rectangle (box car)
+// 			weighingFactor = 1.0f;
+// 			break;
+// 		case FFT_WIN_TYP_HAMMING: // hamming
+// 			weighingFactor = 0.54f - (0.46f * cosf(twoPi * ratio));
+// 			break;
+// 		case FFT_WIN_TYP_HANN: // hann
+// 			weighingFactor = 0.54f * (1.0f - cosf(twoPi * ratio));
+// 			break;
+// 		case FFT_WIN_TYP_TRIANGLE: // triangle (Bartlett)
+// 			#if defined(ESP8266) || defined(ESP32)
+// 			weighingFactor = 1.0 - ((2.0 * fabs(indexMinusOne - (samplesMinusOne / 2.0))) / samplesMinusOne);
+// 			#else
+// 			weighingFactor = 1.0f - ((2.0f * abs(indexMinusOne - (samplesMinusOne / 2.0f))) / samplesMinusOne);
+// 			#endif
+// 			break;
+// 		case FFT_WIN_TYP_NUTTALL: // nuttall
+// 			weighingFactor = 0.355768f - (0.487396f * (cosf(twoPi * ratio))) + (0.144232f * (cosf(fourPi * ratio))) - (0.012604f * (cosf(sixPi * ratio)));
+// 			break;
+// 		case FFT_WIN_TYP_BLACKMAN: // blackman
+// 			weighingFactor = 0.42323f - (0.49755f * (cosf(twoPi * ratio))) + (0.07922f * (cosf(fourPi * ratio)));
+// 			break;
+// 		case FFT_WIN_TYP_BLACKMAN_NUTTALL: // blackman nuttall
+// 			weighingFactor = 0.3635819f - (0.4891775f * (cosf(twoPi * ratio))) + (0.1365995f * (cosf(fourPi * ratio))) - (0.0106411f * (cosf(sixPi * ratio)));
+// 			break;
+// 		case FFT_WIN_TYP_BLACKMAN_HARRIS: // blackman harris
+// 			weighingFactor = 0.35875f - (0.48829f * (cosf(twoPi * ratio))) + (0.14128f * (cosf(fourPi * ratio))) - (0.01168f * (cosf(sixPi * ratio)));
+// 			break;
+// 		case FFT_WIN_TYP_FLT_TOP: // flat top
+// 			weighingFactor = 0.2810639f - (0.5208972f * cosf(twoPi * ratio)) + (0.1980399f * cosf(fourPi * ratio));
+// 			break;
+// 		case FFT_WIN_TYP_WELCH: // welch
+// 			weighingFactor = 1.0f - powf((indexMinusOne - samplesMinusOne / 2.0) / (samplesMinusOne / 2.0), 2);
+// 			break;
+// 		}
+// 		if (dir == FFT_FORWARD) {
+// 			this->_vReal[i] *= weighingFactor;
+// 			this->_vReal[this->_samples - (i + 1)] *= weighingFactor;
+// 		}
+// 		else {
+// 			this->_vReal[i] /= weighingFactor;
+// 			this->_vReal[this->_samples - (i + 1)] /= weighingFactor;
+// 		}
+// 	}
+// }
```

Comment:
We shouldn't merge big blocks of commented code. Are you sure they need to be commented out to reduce the binary size? Wouldn't they just be removed by compiler optimizations?

Reply:
I've just checked : un-commenting those functions do not change the final binary size. The compiler/optimizer are smart enough to remove those functions at build time. So, yes, no need to comment them.

Reply:
@JF002 This library was removed in favor of the template version that @Riksu9000 found.